### PR TITLE
Allow templating `$0` with glob matching

### PR DESCRIPTION
**Pull request by pedro-stanaka**
## Summary

This is my implementation for #438. I went with a naive approach here, I am always adding a new label at `exporter.go` if the metric has not labelset we create a new one and put the new label there.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -6,3 +6,4 @@ dependencies-stamp
 /.tarballs
 *~
 /vendor
+.idea
```
Comment (matthiasr) [inline .gitignore:9]: Please add this to your global .gitignore, not every project

**Review by matthiasr — CHANGES_REQUESTED** [review]
I don't want to make this the default for everyone, and I am uneasy about even hard coding the label value.

As I commented in #438, I found that this used to be possible by mapping the `$0` value into a label (and it still is, if you use the `regex` type match), so I would rather fix that for glob mappings than introduce a new behavior.

**Comment by pedro-stanaka** [comment]
Thanks for the feedback on the issue #438 @matthiasr. I will close that now and try to work with the regex solution you presented. 👍

**Comment by matthiasr** [comment]
Oops got confused 😅

**Pull request closed without merging**